### ISSUE TITLE
ocamlbuild_plugin.cmi is not a compiled interface for this version of oCaml

### ISSUE THREAD

**Issue by dbuttigieg** [ocaml/ocamlbuild#277]
I am currently running OCaml version 4.06.1; ocamlbuild version 0.12.0; oasis version 0.4.11; and I'm trying to build and run my soltution but I'm getting the following error. I am relatively new to OCaml and I simply installed the latest versions using brew (on Mac). 

```Denises-MBP-2:fyp denise$ make
ocaml setup.ml -build
+ ocamlopt.opt unix.cmxa -I /Users/denise/.opam/system/lib/ocamlbuild /Users/denise/.opam/system/lib/ocamlb
uild/ocamlbuildlib.cmxa myocamlbuild.ml /Users/denise/.opam/system/lib/ocamlbuild/ocamlbuild.cmx -o myocaml
build
File "myocamlbuild.ml", line 1:
Error: /Users/denise/.opam/system/lib/ocamlbuild/ocamlbuild_plugin.cmi
is not a compiled interface for this version of OCaml.
It seems to be for an older version of OCaml.
Command exited with code 2.

**Comment by Armael** [comment]
I suspect the issue is that you are using the system switch of opam, which is generally a bad idea (I know, that's what you get by default with opam, which is a shame...). Try creating a new opam switch with `opam switch 4.06.0`.

**Comment by dbuttigieg** [comment]
@Armael  thank you. I shall try. 

Sorry if this question sounds stupid, but once I create the opam switch, do I need to reinstall the libraries/packages?

**Comment by Armael** [comment]
Yes, you will need to reinstall all the packages that you had installed in the system switch.

**Comment by dbuttigieg** [comment]
@Armael  I have configure a new switch and installed the required libraries, after executing  `opam config env --switch 4.06.0` and then `make` I am still getting the same error.

**Comment by Armael** [comment]
Did you run `eval $(opam config env --switch 4.06.0)` ? The `eval $(...)` is needed to correctly setup the environment.

**Comment by dbuttigieg** [comment]
yes, and a new folder within my opam folder was created - then I exported the current libraries and packages from 4.06.1 into 4.06.0, and i ran make.. I am not totally sure however whether the system switch is now the 4.06.0 and not the other 4.06.1..

**Comment by dbuttigieg** [comment]
@Octachron  no I didn't. just did and it worked. thank you so much :)
